### PR TITLE
[Modules] Updated Network/VPNGateways to new dependency approach

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Updated Network/VPNGateways to new dependency approach

| Pipeline |
| - |
| [Network: VPNGateways](https://github.com/Azure/ResourceModules/actions/workflows/ms.network.vpngateways.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,96 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/vpnGateways/.test/min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,47 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.Network/vpnGateways/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    virtualHubResourceId: resourceGroupResources.outputs.virtualHubResourceId
+    bgpSettings: {
+      asn: 65515
+      peerWeight: 0
+    }
+    connections: [
+      {
+        connectionBandwidth: 100
+        enableBgp: false
+        name: 'Connection-dep-<<namePrefix>>-vs-${serviceShort}'
```
Comment (eriqua) [inline modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep:54]: Is the connection name intended to be like `'Connection-<<vpnSiteName>>'`? In that case I'd suggest to create a variable for the vpnSiteName and use that here and as `resourceGroupResources` input

Comment (eriqua) [inline modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep:54]: Is the connection name intended to be like `'Connection-<<vpnSiteName>>'`? In that case I'd suggest to add an output for the `vpnSiteName` from `resourceGroupResources` and use that here

Comment (AlexanderSehr) [reply modules/Microsoft.Network/vpnGateways/.test/common/deploy.test.bicep]: done.

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**